### PR TITLE
Chore: organize test files and fix code coverage

### PR DESCRIPTION
**Pull request by tmontini**
Moved unit tests to /tests/unit folder
Reorganized cypress tests to pass, intermittent and skip folder.
Configured unit tests coverage

### CODE REVIEW

File: app/meteor-accounts-saml/tests/server.tests.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { expect } from 'chai';
 
-import '../../lib/tests/server.mocks.js';
+import '../../../tests/unit/app/lib/server.mocks';
```
Comment (rodrigok) [inline app/meteor-accounts-saml/tests/server.tests.ts:3]: This test is still outside the `tests` folder

File: package.json [hunk 2]
```diff
@@ -353,5 +356,25 @@
...
+			"server/**/*.js",
+			"server/**/*.ts"
+		],
+		"exclude": [
+			"node_modules",
+			"**/.mocharc.js",
+			"**/*.spec.js",
+			"**/*.spec.ts",
+			"**/*.tests.js",
+			"**/*.tests.ts",
```
Comment (rodrigok) [inline package.json:375]: Is this needed since the next line excludes test folder?

**Review by rodrigok — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -353,5 +356,21 @@
...
+			"app/**/*.js",
+			"app/**/*.ts",
+			"ee/**/*.js",
+			"ee/**/*.ts",
+			"server/**/*.js",
+			"server/**/*.ts"
+		],
+		"exclude": [
+			"node_modules",
+			"**/.mocharc.js",
```
Comment (rodrigok) [inline package.json:371]: What about the other mocha config files?

File: package.json [hunk 4]
```diff
@@ -353,5 +356,21 @@
 	"volta": {
 		"node": "14.18.2",
 		"npm": "6.14.15"
+	},
+	"nyc": {
+		"include": [
+			"app/**/*.js",
```
Comment (rodrigok) [inline package.json:362]: Isn't better to include all the folders? Like `**/*.js`?

File: .mocharc.js [hunk 5]
```diff
@@ -23,5 +23,11 @@ Object.assign(
 module.exports = {
 	...base, // see https://github.com/mochajs/mocha/issues/3916
 	exit: true,
-	spec: ['app/**/*.spec.ts', 'app/**/*.tests.js', 'app/**/*.tests.ts', 'server/**/*.tests.ts', 'ee/**/*.tests.ts'],
+	spec: [
+		'ee/**/*.tests.ts',
```
Comment (rodrigok) [inline .mocharc.js:27]: Shouldn't be `'ee/tests/**/*.tests.ts'`?

**Review by rodrigok — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (tmontini) [reply package.json:362]: I tried to run the coverage with `**/*.(js|ts)` but it didn't work

**Review by rodrigok — DISMISSED** [review]

**Review by rodrigok — DISMISSED** [review]

**Review by tassoevan — CHANGES_REQUESTED** [review]
Wouldn't it discourage test maintenance to move them to another root directory instead of keeping them [co-located](https://domhabersack.com/colocating-tests)? Unless there is a tooling limitation, having another file structure will just hurt quality sustenance...

**Review by ggazzo — APPROVED** [review]

**Pull request merged**